server/src/kafka.rs

```diff
@@ -0,0 +1,214 @@
+        cb = cb.with_group(val)
+    }
+    if let Ok(vals_raw) = env::var("KAFKA_PARTITIONS") {
+        let vals = vals_raw
+            .split(',')
+            .map(i32::from_str)
+            .collect::<Result<Vec<i32>, ParseIntError>>();
+        cb = cb.with_topic_partitions(
+            topic,
+            &vals.map_err(|raw| KafkaError::ParseIntError("KAFKA_PARTITIONS", raw))?,
+        );
+    }
+
+    if let Ok(val) = env::var("KAFKA_OFFSET_STORAGE") {
+        cb = cb.with_offset_storage(Some(match val.to_lowercase().as_str() {
+            "kafka" => Ok(GroupOffsetStorage::Kafka),
+            "zookeeper" => Ok(GroupOffsetStorage::Zookeeper),
+            _ => Err(KafkaError::InvalidGroupOffsetStorage(val)),
+        }?))
+    }
+    if get_flag_env_val("KAFKA_USE_SECURITY", false) {
+        let connector = SslConnector::builder(SslMethod::tls()).unwrap().build();
+        let sec = SecurityConfig::new(connector)
+            .with_hostname_verification(get_flag_env_val("KAFKA_SECURITY_VERIFY_HOSTNAME", true));
+        cb = cb.with_security(sec);
+    }
+    if let Ok(val) = env::var("KAFKA_FALLBACK_OFFSET") {
+        cb = cb.with_fallback_offset(match val.to_lowercase().as_str() {
+            "earliest" => Ok(FetchOffset::Earliest),
+            "latest" => Ok(FetchOffset::Latest),
+            val if val.starts_with("by:") => Ok(FetchOffset::ByTime(
+                i64::from_str(&val[3..])
+                    .map_err(|raw| KafkaError::ParseIntError("KAFKA_FALLBACK_OFFSET", raw))?,
+            )),
+            _ => Err(KafkaError::InvalidGroupOffsetStorage(val)),
+        }?)
+    }
+
+    let res = cb.create()?;
+    Ok(res)
+}
+
+fn ingest_message<'a>(stream_name: &str, msg: &Message<'a>) -> Result<(), KafkaError> {
+    log::debug!("Message: {:?}", msg);
+    let hash_map = STREAM_INFO.read().unwrap();
+    let schema = hash_map
+        .get(stream_name)
+        .ok_or(KafkaError::StreamNotFound(stream_name.to_owned()))?
+        .schema
+        .clone();
+
+    let txt = String::from_utf8(msg.value.into_iter().copied().collect());
+    log::debug!("Message text: {}", txt.unwrap());
+    let event = format::json::Event {
+        data: serde_json::from_slice(&msg.value)?,
+        tags: String::default(),
+        metadata: String::default(),
+    };
+    log::debug!("Generated event: {:?}", event.data);
+    let (rb, is_first) = event.into_recordbatch(schema, None, None).unwrap();
+
+    event::Event {
+        rb,
+        stream_name: stream_name.to_string(),
+        origin_format: "json",
+        origin_size: msg.value.len() as u64,
+        is_first_event: is_first,
+        parsed_timestamp: Utc::now().naive_utc(),
+        time_partition: None,
+        custom_partition_values: HashMap::new(),
+        stream_type: StreamType::UserDefined,
+    }
+    .process_unchecked()?;
+    Ok(())
+}
+
+pub async fn setup_integration() -> Result<JoinHandle<()>, KafkaError> {
```

Comment:
It should be fine to drop the handle if I am not wrong, why are we returning it?